lib/dotcom/system_status/groups.ex

```diff
@@ -0,0 +1,221 @@
+defmodule Dotcom.SystemStatus.Groups do
+  @moduledoc """
+  A module that groups alerts into statuses for the system status widget.
+  """
+
+  alias Alerts.Alert
+
+  @lines ["Blue", "Orange", "Red", "Green"]
+  @green_line_branches ["Green-B", "Green-C", "Green-D", "Green-E"]
+  @routes ["Blue", "Orange", "Red", "Mattapan"] ++ @green_line_branches
+
+  def groups(alerts, now) do
+    grouped_alerts = Map.new(@routes, &{&1, alerts_for_line(alerts, &1)})
+
+    @routes
+    |> Enum.map(fn route_id ->
+      statuses =
+        grouped_alerts
+        |> Map.get(route_id)
+        |> alerts_to_statuses(now)
+        |> consolidate_duplicate_descriptions()
+        |> sort_statuses()
+        |> stringify_times()
+        |> maybe_add_now_text()
+
+      %{route_id: route_id, sub_routes: [], statuses: statuses}
+    end)
+    |> combine_green_line_branches()
+    |> combine_mattapan_with_red_line()
+    |> sort_routes_and_sub_routes()
+  end
+
+  defp alerts_for_line(alerts, line_id) do
+    alerts
+    |> Enum.filter(fn %Alert{informed_entity: informed_entity} ->
+      informed_entity
+      |> Enum.any?(fn
+        %{route: ^line_id} -> true
+        %{} -> false
+      end)
+    end)
+  end
+
+  defp alerts_to_statuses([], _now) do
+    [%{description: "Normal Service", time: nil}]
+  end
+
+  defp alerts_to_statuses(alerts, now) do
+    alerts
+    |> Enum.map(fn alert ->
+      alert_to_status(alert, now)
+    end)
+  end
+
+  defp alert_to_status(alert, now) do
+    description =
+      case alert.effect do
+        :delay -> "Delays"
+        :shuttle -> "Shuttle Buses"
+        :station_closure -> "Station Closure"
+        :suspension -> "Suspension"
+      end
+
+    time = future_start_time(alert.active_period, now)
+
+    %{description: description, time: time}
+  end
+
+  defp stringify_times(statuses) do
+    statuses
+    |> Enum.map(fn status ->
+      case status do
+        %{time: nil} ->
+          status
+
+        %{time: time} ->
+          %{status | time: Timex.format!(time, "%-I:%M%p", :strftime) |> String.downcase()}
```

Comment:
I've seen this time format multiple places. Maybe it is a good time to turn it into a function and move it to a util module.

Reply:
https://github.com/mbta/dotcom/pull/2337